README.md

```diff
@@ -58,7 +59,7 @@ Are you looking for 0.x docs? Find them [here](https://github.com/jquery/globali
 <a name="about"></a>
 ## About Globalize
 
-<a name="why"></a>
+<a name="why_globalization"></a>
```

Comment:
Since, we'll eventually migrate README into api.globalizejs.com, I'm not too concerned with this for now.